src/main/java/com/wavefront/dropwizard/metrics/DropwizardMetricsReporter.java

```diff
@@ -250,16 +251,25 @@ private DropwizardMetricsReporter(MetricRegistry registry,
     this.histogramGranularities = histogramGranularities;
 
     if (includeJvmMetrics) {
-      registry.register("jvm.uptime",
+      tryRegister(registry, "jvm.uptime",
           (Gauge<Long>) () -> ManagementFactory.getRuntimeMXBean().getUptime());
-      registry.register("jvm.current_time", (Gauge<Long>) clock::getTime);
-      registry.register("jvm.classes", new ClassLoadingGaugeSet());
-      registry.register("jvm.fd_usage", new FileDescriptorRatioGauge());
-      registry.register("jvm.buffers",
+      tryRegister(registry, "jvm.current_time", (Gauge<Long>) clock::getTime);
+      tryRegister(registry, "jvm.classes", new ClassLoadingGaugeSet());
+      tryRegister(registry, "jvm.fd_usage", new FileDescriptorRatioGauge());
+      tryRegister(registry, "jvm.buffers",
           new BufferPoolMetricSet(ManagementFactory.getPlatformMBeanServer()));
-      registry.register("jvm.gc", new GarbageCollectorMetricSet());
-      registry.register("jvm.memory", new MemoryUsageGaugeSet());
-      registry.register("jvm.thread-states", new ThreadStatesGaugeSet());
+      tryRegister(registry, "jvm.gc", new GarbageCollectorMetricSet());
+      tryRegister(registry, "jvm.memory", new MemoryUsageGaugeSet());
+      tryRegister(registry, "jvm.thread-states", new ThreadStatesGaugeSet());
+    }
+  }
+
+  private <T extends Metric> void tryRegister(MetricRegistry registry, String name, T metric) {
+    // Dropwizard services automatically include JVM metrics, so adding them again would throw an exception
+    try {
+      registry.register(name, metric);
+    } catch (IllegalArgumentException e) {
+      logger.log(Level.INFO, "Metric with the same name already exists", e);
```

Comment:
No need to emit `e`
but please include the name of the metric. We don't developers who use our SDKs to see exception stack traces in the apps.

Reply:
Changed to `e.getMessage()` because `name` may not be the full metric name if registering a MetricSet.